dags/hivemind_etl_helpers/discord_query.py

```diff
@@ -0,0 +1,121 @@
+from hivemind_etl_helpers.src.retrievers.forum_summary_retriever import (
+    ForumBasedSummaryRetriever,
+)
+from hivemind_etl_helpers.src.utils.cohere_embedding import CohereEmbedding
+from hivemind_etl_helpers.src.utils.pg_vector_access import PGVectorAccess
+from llama_index import QueryBundle
+from llama_index.vector_stores import ExactMatchFilter, FilterCondition, MetadataFilters
+
+
+def query_discord(
+    community_id: str,
+    query: str,
+    thread_names: list[str],
```

Comment:
If no thread_names / channel_names / days are provided, does the search cover all (does it not apply any filtering)? Also, is it possible to add an author filter too (again, all authors are considered when no filters are specified)?

Reply:
And when an author filter is added, no summary search should be done (see card description)

Reply:
I have written two functions 
- `query_discord_auto_filter` will do a first summary search, then would extract the metadata information from the summaries.
- `query_discord` will use the given filters (from the summary metadata or just the user given in discord-bot), and fetch the results from raw-data.

In case of no thread_names/channel_names/days given the for loops within the function `query_discord` wouldn't happen and we would do no filtering on raw data.

If we want no summary search to be done we would just go for the `query_discord` function, else, we would go with `query_discord_auto_filter` function.

These codes will be moved to a repository named `hivemind-bot` but they are the utility functions we would use to  query the raw or/and the summary data.

Reply:
Sounds good!